infocmdb/ci_type.go

```diff
@@ -143,3 +144,139 @@ func (c *Client) SetTypeOfCi(ciId int, ciType string) (err error) {
 
 	return
 }
+
+type respCreateCiType struct {
+	Success bool         `json:"success"`
+	Message string       `json:"message"`
+	Data    []responseId `json:"data"`
+}
+
+type CiTypeParams struct {
+	Name                    string
+	Description             string
+	Note                    string
+	ParentCiTypeId          int
+	OrderNumber             int
+	createButtonDescription string
+	icon                    string
+	query                   string
+	DefaultProjectId        int
+	defaultAttributeId      int
+	defaultSortAttributeId  int
+	isDefaultSortAsc        int
+	IsCiAttach              int
+	IsAttributeAttach       int
+	tag                     string
+	isTabEnabled            int
+	isEventEnabled          int
+	IsActive                int
+	userId                  int
+}
+
+func (c *Client) NewCiTypeParams() (params *CiTypeParams) {
+	params = &CiTypeParams{
+		Name:                    "",
+		Description:             "",
+		Note:                    "",
+		ParentCiTypeId:          0,
+		OrderNumber:             0,
+		createButtonDescription: "",
+		icon:                    "",
+		query:                   "",
+		DefaultProjectId:        0,
+		defaultAttributeId:      0,
+		defaultSortAttributeId:  0,
+		isDefaultSortAsc:        0,
+		IsCiAttach:              0,
+		IsAttributeAttach:       0,
+		tag:                     "",
+		isTabEnabled:            0,
+		isEventEnabled:          0,
+		IsActive:                1,
+		userId:                  0,
+	}
+	return
+}
+
+func (c *Client) CreateCiType(typeParams *CiTypeParams) (typeId int, err error) {
+
+	if err = c.v2.Login(); err != nil {
+		return
+	}
+
+	typeIdExists, _ := c.GetCiTypeIdByCiTypeName(typeParams.Name)
+
+	if typeIdExists == 0 {
```

Comment:
I usually use a early return here to remove the extra indention in the code.
```golang
if condition {
    return ...
}

// else... other non-indented code
```

*This usually only matters if there are more nested if conditions and I wouldn't have commented on that if there wouldn't be other required changes.
Sorry for the nitpick.*

Reply:
Should be ok now.